executor/simple_test.go

```diff
@@ -871,3 +871,15 @@ func (s *testSuite3) TestIssue17247(c *C) {
 	_, err := tk1.Exec("ALTER USER USER() IDENTIFIED BY PASSWORD '*B50FBDB37F1256824274912F2A1CE648082C3F1F'")
 	c.Assert(err, NotNil)
 }
+
+// Close issue #23649.
+// See https://github.com/pingcap/tidb/issues/23649
+func (s *testSuite3) TestIssue23649(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("DROP USER IF EXISTS issue23649;")
+	tk.MustExec("CREATE USER issue23649;")
+	_, err := tk.Exec("GRANT bogusrole to issue23649;")
+	c.Assert(err.Error(), Equals, "[executor:3523]Unknown authorization ID `bogusrole`@`%`")
+	_, err = tk.Exec("GRANT bogusrole to nonexisting;")
+	c.Assert(err.Error(), Equals, "[executor:3523]Unknown authorization ID `bogusrole`@`%`")
```

Comment:
well, this test result is different from the issue. I have not verified it by myself, please check it again.

Reply:
i execute in mysql, and get **ERROR 3523 (HY000): Unknown authorization ID `bogusrole`@`%`** twice

Reply:
great, thank you